nautobot_secrets_providers/providers/azure.py

```diff
@@ -0,0 +1,70 @@
+"""Secrets Provider for Azure Key Vault."""
+
+import json
+
+try:
+  from azure.identity import DefaultAzureCredential
+  from azure.keyvault.secrets import SecretClient
+  azure_available = True
+except (ImportError, ModuleNotFoundError):
+  azure_available = False
+
+from django import forms
+from nautobot.core.forms import BootstrapMixin
+from nautobot.extras.secrets import exceptions, SecretsProvider
+
+__all__ = ("AzureKeyVaultSecretsProvider")
+
+class AzureKeyVaultSecretsProvider(SecretsProvider):
+    """A secrets provider for Azure Key Vault."""
+
+    slug = "azure-key-vault"
+    name = "Azure Key Vault"
+    is_available = azure_available
+    
+    # pylint: disable-next=nb-incorrect-base-class
+    class ParametersForm(BootstrapMixin, forms.Form):
+        """Required parameters for Azure Key Vault."""
+
+        vault_url = forms.CharField(
+            required=True,
+            help_text="The URL of the Azure Key Vault",
+        )
+        secret_name = forms.CharField(
+            required=True,
+            help_text="The name of the secret in the Azure Key Vault",
+        )
```

Comment:
Sorry I am not super familiar with the secrets provider, how/where are we using the `ParametersForm`?

Reply:
It's for this Parameters form section when adding a new secret. It updates the form fields dynamically based on what provider you select. Believe it relates to this code in core: https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/secrets/providers.py 
![image](https://github.com/nautobot/nautobot-app-secrets-providers/assets/111259311/aa7e8648-6e6d-4e4a-9c55-d1cca782b88e)